### DISCUSSION TITLE
Weight column missing

### DISCUSSION THREAD

**Discussion by ratheraarif** [open2c/cooler#377] [Q&A] [answered]
HI,
I need to run the following code from cooltools

`# cvd == contacts-vs-distance
cvd = cooltools.expected_cis(
    clr=clr,
    view_df=hg38_arms,
    smooth=False,
    aggregate_smoothed=False,
    nproc=4 #if you do not have multiple cores available, set to 1
)`

However it flags the error `cooler is not balanced, or balancing weight weight is not available in the cooler`
Note that I generated the mcool files from .hic data using the following code hic2cool convert file1.allValidPairs.hic file2 -r 0 , which probably misses the `weight ` column.  I can balance the matrix through `cooler.balance_cooler(clr)`, however, I cannot add it to the bins.

Please help me to fix this issue

**Comment by Phlya** [comment]
https://cooltools.readthedocs.io/en/latest/cooltools.html#cooltools.api.expected.expected_cis
This function has a clr_weight_name argument, with coolers produced from .hic files specify an existing weight, e.g. `clr_weight_name='KR'` and it should work.
Same should apply to all functions in cooltools.

**Comment by ratheraarif** [comment]
I did the same but it still generates the error..

<img width="1471" alt="image" src="https://github.com/open2c/cooler/assets/46924431/99510535-fe3c-4542-b7e3-599ab0aed3a6">

**Comment by Phlya** [comment]
Right, ok. Then just use `cooler balance` command in the terminal, that's the easiest way to balance probably.

**Comment by ratheraarif** [comment]
Thank you for your reply,

I can see that I can create a balancing using `cooler balance` ,but this lets me balance each resolution separately. Is there a way to balance the entire mcool file simultaneously, irrespective of the number of resolutions..
Please help with the code example

**Comment by Phlya** [comment, marked as answer]
No, just run `cooler balance` for each resolution.
Here is a bash one-liner for that:
```
for i in `cooler ls <file.mcool>`;  do cooler balance $i; done;
```
